feat(monitor): monitor the chroot systemcall

haytok opened the pull request:
Currently, KubeArmor does not have the feature to monitor the chroot
system call.

On the other hand, there is a future request in Issue #1070 to support
audit for security sensitive system calls.

  - Issue #1070 

Therefore, this commit implements a new feature to monitor the chroot
system call.

This commit also includes a corresponding fix to allow standard input to
be used in the `K8sExecInPod function` in `kartutil.go` to run ginkgo test for
the chroot system call and so on.

**Purpose of PR?**:

Fixes #1070 

**Does this PR introduce a breaking change?**

No

**If the changes in this PR are manually verified, list down the scenarios covered:**:

The `chroot` system call can be monitored

**Additional information for reviewer?** :
_Mention if this PR is part of any design or a continuation of previous PRs_

- Result of `karmor logs --json` confirming that the `chroot` system call has been executed

```json
{
  "Timestamp": 1689491852,
  "UpdatedTime": "2023-07-16T07:17:32.077185Z",
  "ClusterName": "default",
  "HostName": "ip-172-31-36-15",
  "NamespaceName": "syscalls",
  "Owner": {
    "Ref": "Deployment",
    "Name": "ubuntu-1-deployment",
    "Namespace": "syscalls"
  },
  "PodName": "ubuntu-1-deployment-9c9dbdb8-mpkzg",
  "Labels": "container=ubuntu-1",
  "ContainerID": "fc231ca71c13188feb32aaf7c41f6357d2576411dba14a8a913119a2e70c3ea8",
  "ContainerName": "ubuntu-1-container",
  "ContainerImage": "kubearmor/ubuntu-w-utils:0.1@sha256:b4693b003ed1fbf7f5ef2c8b9b3f96fd853c30e1b39549cf98bd772fbd99e260",
  "HostPPID": 3392448,
  "HostPID": 3392770,
  "PPID": 3392448,
  "PID": 96,
  "ParentProcessName": "/usr/bin/containerd-shim",
  "ProcessName": "/bin/sh",
  "PolicyName": "audit-all-chroot",
  "Severity": "3",
  "Type": "MatchedPolicy",
  "Source": "/bin/sh -i",
  "Operation": "Syscall",
  "Resource": "/",
  "Data": "syscall=SYS_CHROOT",
  "Action": "Audit",
  "Result": "Passed"
}
```

- Results of tests by `Ginkgo`

The newly created scenario passed.

```bash
Created a gRPC client (localhost:32767)
Checked the liveness of the gRPC server
Started to watch alerts
releasing grpc client
Stopped WatchAlerts
•
```

**Checklist:**
- [ ] Bug fix. Fixes #<issue number>
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [x] PR Title follows the convention of  `<type>(<scope>): <subject>`
- [ ] Commit has unit tests
- [ ] Commit has integration tests

<!--

The PR title message must follow convention:
`<type>(<scope>): <subject>`.

Where: <br />
- `type` is to define what type of PR is this.
  Most common types are:
    - `feat`      - for new features, not a new feature for build script
    - `fix`       - for bug fixes or improvements, not a fix for build script
    - `chore`     - changes not related to production code
    - `docs`      - changes related to documentation
    - `style`     - formatting, missing semi colons, linting fix etc; no significant production code changes
    - `test`      - adding missing tests, refactoring tests; no production code change
    - `refactor`  - refactoring production code, eg. renaming a variable or function name, there should not be any significant production code changes

- `scope` is a single word that best describes where the changes fit.
    - feature(`monitor`,`enforcer`)
    - test(`tests`, `bdd`)
    - chore(`build`)
- `subject` is a single line brief description of the changes made in the pull request.

-->

daemon1024 reviewed (changes requested):
Thanks for the PR. I think we can include this, really appreciate you including the tests as well. Can you rebase the PR and handle conflicts?


I see that karmor.te is also changed, I don't it should be updated for these changes. Please clarify.

haytok committed:
feat(monitor): monitor the chroot systemcall

Currently, KubeArmor does not have the feature to monitor the chroot
system call.

On the other hand, there is a future request in Issue #1070 to support
audit for security sensitive system calls.

  - Issue #1070 : support audit for security sensitive system calls
    - #1070

Therefore, this commit implements a new feature to monitor the chroot
system call.

This commit also includes a corresponding fix to allow standard input to
be used in the K8sExecInPod function in kartutil.go to run ginkgo test for
the chroot system call and so on.

Signed-off-by: Hayato Kiwata <haytok@amazon.co.jp>

haytok committed:
feat(monitor): monitor the chroot systemcall

Currently, CI tests are time out and failing.

Therefore, in this commit, we will fix it so that the test timeout period
is extended.

Signed-off-by: Hayato Kiwata <haytok@amazon.co.jp>

haytok committed:
feat(monitor): monitor the chroot systemcall

Currently, CI tests are time out and failing.

Therefore, in this commit, we will fix it so that the test timeout period
is extended.

Signed-off-by: Hayato Kiwata <haytok@amazon.co.jp>
